specification/applicationinsights/data-plane/microsoft.insights/preview/2018-04-20/swagger.json

```diff
@@ -0,0 +1,2071 @@
+        },
+        "parameters": [
+          {
+            "$ref": "#/parameters/subscriptionId"
+          },
+          {
+            "$ref": "#/parameters/resourceGroupName"
+          },
+          {
+            "$ref": "#/parameters/applicationName"
+          },
+          {
+            "$ref": "#/parameters/queryBody"
+          },
+          {
+            "$ref": "#/parameters/apiVersion"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Successful response",
+            "schema": {
+              "$ref": "#/definitions/queryResults"
+            }
+          },
+          "default": {
+            "description": "An error response object.",
+            "schema": {
+              "$ref": "#/definitions/errorResponse"
+            }
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourcegroups/{resourceGroupName}/providers/Microsoft.Insights/components/{applicationName}/metrics/{metricId}": {
+      "get": {
+        "operationId": "GetMetric",
+        "summary": "Retrieve metric data",
+        "description": "Gets metric values for a single metric",
+        "x-ms-examples": {
+          "simpleMetric": {
+            "$ref": "examples/metric-get.json"
+          },
+          "intervaledMetric": {
+            "$ref": "examples/metric-get-full.json"
+          }
+        },
+        "parameters": [
+          {
+            "$ref": "#/parameters/subscriptionId"
+          },
+          {
+            "$ref": "#/parameters/resourceGroupName"
+          },
+          {
+            "$ref": "#/parameters/applicationName"
+          },
+          {
+            "$ref": "#/parameters/metricId"
+          },
+          {
+            "$ref": "#/parameters/metricsTimespan"
+          },
+          {
+            "$ref": "#/parameters/metricsInterval"
+          },
+          {
+            "$ref": "#/parameters/metricsAggregation"
+          },
+          {
+            "$ref": "#/parameters/metricsSegment"
+          },
+          {
+            "$ref": "#/parameters/metricsTop"
+          },
+          {
+            "$ref": "#/parameters/metricsOrderBy"
+          },
+          {
+            "$ref": "#/parameters/metricsFilter"
+          },
+          {
+            "$ref": "#/parameters/apiVersion"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Successful response",
+            "schema": {
+              "$ref": "#/definitions/metricsResult"
+            }
+          },
+          "default": {
+            "description": "An error response object.",
+            "schema": {
+              "$ref": "#/definitions/errorResponse"
+            }
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourcegroups/{resourceGroupName}/providers/Microsoft.Insights/components/{applicationName}/metrics/metadata": {
+      "get": {
+        "operationId": "GetMetricsMetadata",
```

Comment:
Given that you broke out "Query" into its own operation group, I'd recommend doing the same for Metric and Event. That would mean that you'd have:
- `Metric_GetMetadata`
- `Metric_Get`
- `Event_List`
- `Event_Get`
- `Event_GetMetadata` (optionally with the OData Suffix as well, based on your judgement.)

Reply:
This is great -- I was hoping to do some cleanup here but had only played around with /query. Can you elaborate on how this impacts things downstream from swagger? From what I can tell this is used primarily for grouping operations in docs and (maybe in some of the?) SDKs? Still exploring how all the pieces fit together 😄

Reply:
Ah yeah, for sure! The main difference as you suspect, is about group operations together. In addition to doc generation impacts, the AutoRest modeler actually assigns each operation group its own client. For example, you can see in the Go SDK for the Compute there are separate clients for [interacting with VirtualMachines](https://github.com/Azure/azure-sdk-for-go/blob/0a8ed6c65dc34935d90dee3ddbc901f1f497f221/services/compute/mgmt/2018-04-01/compute/virtualmachines.go#L29) and [interacting with a VM Image](https://github.com/Azure/azure-sdk-for-go/blob/0a8ed6c65dc34935d90dee3ddbc901f1f497f221/services/compute/mgmt/2018-04-01/compute/virtualmachineimages.go#L28).

Getting the balance right of not making people new-up clients for everything, but also keeping operations discoverable is a bit of an art. You're really the only one who can tell me if it makes sense to interact with Queries, Metrics, and Events at the same time, or if there's always separation there. 😃

Reply:
Hmm...So I grouped the operations up and generated the C# and Python SDKs for comparison. I see how the operations group but It looks like they don't get a new client? Using the compute example, [here's the C# version](https://github.com/Azure/azure-sdk-for-net/blob/psSdkJson6/src/SDKs/Compute/Management.Compute/Generated/VirtualMachineImagesOperations.cs). I think for our purposes one client makes sense anyway, but I couldn't replicate what the Go SDK you pointed out was doing in another language -- wanted to clarify the expected behavior?